File: workflows-java/src/main/java/com/template/flows/GetDealsByAccountFlow.java
package com.template.flows;

import co.paralleluniverse.fibers.Suspendable;
import com.template.states.AccountsDealState;
import net.corda.core.contracts.StateAndRef;
import net.corda.core.flows.FlowException;
import net.corda.core.flows.FlowLogic;
import net.corda.core.flows.InitiatingFlow;
import net.corda.core.flows.StartableByRPC;
import net.corda.core.node.services.Vault;
import net.corda.core.node.services.vault.QueryCriteria;
import net.corda.core.node.services.vault.QueryCriteriaUtils;

import java.util.List;
import java.util.UUID;

@InitiatingFlow
@StartableByRPC
public class GetDealsByAccountFlow extends FlowLogic<List<StateAndRef<AccountsDealState>>> {
    private final List<UUID> accountUUIDs;

    public GetDealsByAccountFlow(List<UUID> accountUUIDs) {
        this.accountUUIDs = accountUUIDs;
    }

    @Suspendable
    @Override
    public List<StateAndRef<AccountsDealState>> call() throws FlowException {
        int DEFAULT_PAGE_SIZE = QueryCriteriaUtils.DEFAULT_PAGE_SIZE;

        Vault.Page<AccountsDealState> results = getServiceHub().getVaultService().queryBy(
                AccountsDealState.class,
                new QueryCriteria.VaultQueryCriteria().withExternalIds(accountUUIDs)
        );
        if (results.getTotalStatesAvailable() > DEFAULT_PAGE_SIZE)
            throw new FlowException("more results than max page size");
        return results.getStates();
    }
}

File: workflows-kotlin/src/main/kotlin/com/template/flows/GetDealsByAccountFlow.kt
package com.template.flows

import co.paralleluniverse.fibers.Suspendable
import com.template.states.AccountDealState
import net.corda.core.contracts.StateAndRef
import net.corda.core.flows.FlowException
import net.corda.core.flows.FlowLogic
import net.corda.core.flows.InitiatingFlow
import net.corda.core.flows.StartableByRPC
import net.corda.core.node.services.queryBy
import net.corda.core.node.services.vault.DEFAULT_PAGE_NUM
import net.corda.core.node.services.vault.QueryCriteria
import java.security.PublicKey
import java.util.*

@InitiatingFlow
@StartableByRPC
class GetDealsByAccountFlow(val accountUUIDs: List<UUID>): FlowLogic<List<StateAndRef<AccountDealState>>>() {

    @Suspendable
    override fun call():List<StateAndRef<AccountDealState>> {


        val results = serviceHub.vaultService.queryBy(
                AccountDealState::class.java,
                QueryCriteria.VaultQueryCriteria(externalIds = accountUUIDs)

        )
        if (results.totalStatesAvailable > DEFAULT_PAGE_NUM) throw FlowException("More results than max page size")

    return results.states

    }

}